### ISSUE TITLE
Make response of Interceptor always asynchronous

### ISSUE THREAD

**Issue by Rhobal** [nigrosimone/ng-http-caching#5]
The current behaviour is that with a cache miss, the subscription will be executed asynchronously, but a cache hit will cause a synchronous trigger of the subscription. The request is that the returned Observable is always async

**Proposed Fix:** 
I think the fix should be simple: change line 34 in https://github.com/nigrosimone/ng-http-caching/blob/master/projects/ng-http-caching/src/lib/ng-http-caching-interceptor.service.ts to 
```typescript
      return scheduled(cachedResponse.clone(), asapScheduler);
```

**More information**

Without caching, it's always asynchronous; if I add ng-http-caching, I also have to make sure that my code works with synchronous execution, making my life harder. I basically have to duplicate my test to run simulate async cache-miss mode and sync cache-hit mode. Or I have to add another HttpInterceptor to `delay(0)` every request.

The behaviour can be demonstrated by adding some logging to the demo:
```typescript
    console.log('pre-request');
    this.http.get(this.url, { headers }).subscribe((result) => {
      this.timeSpan = new Date().getTime() - timeStart.getTime();
      this.updateCachedKeys();
      console.log('response', result);
    });
    console.log('sent request');
```

Executing the request twice gives the following sequence:
```
pre-request
sent request
response

pre-request
response
sent request
```

**Comment by nigrosimone** [comment]
Thanks... you are right, but the fix is 

```
return scheduled(of(cachedResponse.clone()), asapScheduler);
```

scheduled require an Observable

Fixed in version 15.0.2 now on NPM

**Comment by Rhobal** [comment]
Thanks, great support!
